libEGL warning: failed to open /dev/dri/renderD128: Permission denied

libEGL warning: NEEDS EXTENSION: falling back to kms_swrast
libEGL warning: MESA-LOADER: failed to open kms_swrast: /usr/lib/dri/kms_swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)

libEGL warning: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)

libEGL warning: MESA-LOADER: failed to open swrast: /usr/lib/dri/swrast_dri.so: cannot open shared object file: No such file or directory (search paths /usr/lib/x86_64-linux-gnu/dri:\$${ORIGIN}/dri:/usr/lib/dri, suffix _dri)

Traceback (most recent call last):
  File "/home/michael_bezick/Repos/C-LAIfO/train_LAIL_MI.py", line 4, in <module>
    import point_cloud_generator
  File "/home/michael_bezick/Repos/C-LAIfO/point_cloud_generator.py", line 6, in <module>
    from dm_control.suite.walker import Physics
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/suite/__init__.py", line 24, in <module>
    from dm_control.suite import acrobot
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/suite/acrobot.py", line 20, in <module>
    from dm_control import mujoco
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/mujoco/__init__.py", line 18, in <module>
    from dm_control.mujoco.engine import action_spec
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/mujoco/engine.py", line 41, in <module>
    from dm_control import _render
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/_render/__init__.py", line 67, in <module>
    Renderer = import_func()  # pylint: disable=invalid-name
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/_render/__init__.py", line 36, in _import_egl
    from dm_control._render.pyopengl.egl_renderer import EGLContext
  File "/home/michael_bezick/miniconda3/envs/combined/lib/python3.9/site-packages/dm_control/_render/pyopengl/egl_renderer.py", line 75, in <module>
    raise ImportError('Cannot initialize a headless EGL display.')
ImportError: Cannot initialize a headless EGL display.
